### ISSUE TITLE
Run ansible-lint as part of CI linter job

### ISSUE THREAD

**Issue by Spredzy** [ansible/ansible-collection-memsource#35]
Today 1.0.1 was successfully uploaded to galaxy.ansible.com.

While import was running several lint isssues came up, ansibe-lint should be run as part of the linter job this way we ensure clean upload.

Trace

```Starting import: task_id=21384, artifact_id=21386 
Importing with galaxy-importer 0.4.0 
Getting doc strings via ansible-doc 
Finding content inside collection 
Loading role post_translation 
Linting role post_translation via ansible-lint... 
[34mroles/post_translation/defaults/main.yml[0m:12: [91myaml[0m [2mno new line character at the end of file[0m [2;91m(new-line-at-end-of-file)[0m 
[34mroles/post_translation/meta/main.yml[0m:4: [91myaml[0m [2mtrailing spaces[0m [2;91m(trailing-spaces)[0m 
[34mroles/post_translation/meta/main.yml[0m:5: [91myaml[0m [2mtrailing spaces[0m [2;91m(trailing-spaces)[0m 
[34mroles/post_translation/meta/main.yml[0m:11: [91myaml[0m [2mtrailing spaces[0m [2;91m(trailing-spaces)[0m 
[34mroles/post_translation/tasks/cleanup.yml[0m:6: [91myaml[0m [2mno new line character at the end of file[0m [2;91m(new-line-at-end-of-file)[0m 
[34mroles/post_translation/tasks/download_from_memsource.yml[0m:7: [91mrisky-file-permissions[0m [2mFile permissions unset or incorrect[0m 
[34mroles/post_translation/tasks/download_from_memsource.yml[0m:15: [91myaml[0m [2mtrailing spaces[0m [2;91m(trailing-spaces)[0m 
[34mroles/post_translation/tasks/download_from_memsource.yml[0m:25: [91myaml[0m [2mtrailing spaces[0m [2;91m(trailing-spaces)[0m 
[34mroles/post_translation/tasks/download_from_memsource.yml[0m:35: [91munnamed-task[0m [2mAll tasks should be named[0m 
[34mroles/post_translation/tasks/download_from_memsource.yml[0m:38: [91mrisky-file-permissions[0m [2mFile permissions unset or incorrect[0m 
[34mroles/post_translation/tasks/download_from_memsource.yml[0m:44: [91mno-log-password[0m [2mpassword should not be logged.[0m 
[34mroles/post_translation/tasks/download_from_memsource.yml[0m:53: [91myaml[0m [2mno new line character at the end of file[0m [2;91m(new-line-at-end-of-file)[0m 
[34mroles/post_translation/tasks/main.yml[0m:10: [91myaml[0m [2mtrailing spaces[0m [2;91m(trailing-spaces)[0m 
[34mroles/post_translation/tasks/main.yml[0m:11: [91myaml[0m [2mtrailing spaces[0m [2;91m(trailing-spaces)[0m 
[34mroles/post_translation/tasks/main.yml[0m:17: [91myaml[0m [2mtrailing spaces[0m [2;91m(trailing-spaces)[0m 
[34mroles/post_translation/tasks/main.yml[0m:20: [91myaml[0m [2mtrailing spaces[0m [2;91m(trailing-spaces)[0m 
[34mroles/post_translation/tasks/main.yml[0m:26: [91myaml[0m [2mtrailing spaces[0m [2;91m(trailing-spaces)[0m 
[34mroles/post_translation/tasks/main.yml[0m:27: [91myaml[0m [2mno new line character at the end of file[0m [2;91m(new-line-at-end-of-file)[0m 
[34mroles/post_translation/tasks/move_translated_strings.yml[0m:7: [91mrisky-file-permissions[0m [2mFile permissions unset or incorrect[0m 
[34mroles/post_translation/tasks/move_translated_strings.yml[0m:13: [91mvar-spacing[0m [2mVariables should have spaces before and after: {{clone_directory.path}}[0m 
[34mroles/post_translation/tasks/move_translated_strings.yml[0m:21: [91mno-changed-when[0m [2mCommands should not change things if nothing needs doing[0m 
[34mroles/post_translation/tasks/move_translated_strings.yml[0m:21: [91myaml[0m [2mtrailing spaces[0m [2;91m(trailing-spaces)[0m 
[34mroles/post_translation/tasks/move_translated_strings.yml[0m:24: [91myaml[0m [2mtrailing spaces[0m [2;91m(trailing-spaces)[0m 
[34mroles/post_translation/tasks/move_translated_strings.yml[0m:25: [91myaml[0m [2mtrailing spaces[0m [2;91m(trailing-spaces)[0m 
[34mroles/post_translation/tasks/move_translated_strings.yml[0m:28: [91mcommand-instead-of-module[0m [2mgit used in place of git module[0m 
[34mroles/post_translation/tasks/move_translated_strings.yml[0m:28: [91mno-changed-when[0m [2mCommands should not change things if nothing needs doing[0m 
[34mroles/post_translation/tasks/move_translated_strings.yml[0m:37: [91mrisky-file-permissions[0m [2mFile permissions unset or incorrect[0m 
[34mroles/post_translation/tasks/move_translated_strings.yml[0m:42: [91mno-changed-when[0m [2mCommands should not change things if nothing needs doing[0m 
[34mroles/post_translation/tasks/move_translated_strings.yml[0m:49: [91myaml[0m [2mwrong indentation: expected 4 but found 2[0m [2;91m(indentation)[0m 
[34mroles/post_translation/tasks/move_translated_strings.yml[0m:54: [91mcommand-instead-of-shell[0m [2mUse shell only when shell functionality is required[0m 
[34mroles/post_translation/tasks/push_strings.yml[0m:3: [91mcommand-instead-of-module[0m [2mgit used in place of git module[0m 
[34mroles/post_translation/tasks/push_strings.yml[0m:3: [91mno-changed-when[0m [2mCommands should not change things if nothing needs doing[0m 
[34mroles/post_translation/tasks/push_strings.yml[0m:12: [91myaml[0m [2mtrailing spaces[0m [2;91m(trailing-spaces)[0m 
[34mroles/post_translation/tasks/push_strings.yml[0m:14: [91mno-changed-when[0m [2mCommands should not change things if nothing needs doing[0m 
[34mroles/post_translation/tasks/push_strings.yml[0m:15: [91myaml[0m [2mtrailing spaces[0m [2;91m(trailing-spaces)[0m 
[34mroles/post_translation/tasks/push_strings.yml[0m:18: [91myaml[0m [2mtoo few spaces before comment[0m [2;91m(comments)[0m 
[34mroles/post_translation/tasks/push_strings.yml[0m:19: [91myaml[0m [2mtoo few spaces before comment[0m [2;91m(comments)[0m 
[34mroles/post_translation/tasks/push_strings.yml[0m:21: [91myaml[0m [2mno new line character at the end of file[0m [2;91m(new-line-at-end-of-file)[0m 
Loading role pre_translation 
Linting role pre_translation via ansible-lint... 
[34mroles/pre_translation/defaults/main.yml[0m:10: [91myaml[0m [2mno new line character at the end of file[0m [2;91m(new-line-at-end-of-file)[0m 
[34mroles/pre_translation/defaults/main.yml[0m:10: [91myaml[0m [2mtrailing spaces[0m [2;91m(trailing-spaces)[0m 
[34mroles/pre_translation/meta/main.yml[0m:4: [91myaml[0m [2mtrailing spaces[0m [2;91m(trailing-spaces)[0m 
[34mroles/pre_translation/meta/main.yml[0m:11: [91myaml[0m [2mtrailing spaces[0m [2;91m(trailing-spaces)[0m 
[34mroles/pre_translation/tasks/cleanup.yml[0m:6: [91myaml[0m [2mno new line character at the end of file[0m [2;91m(new-line-at-end-of-file)[0m 
[34mroles/pre_translation/tasks/extract_strings.yml[0m:3: [91mrisky-file-permissions[0m [2mFile permissions unset or incorrect[0m 
[34mroles/pre_translation/tasks/extract_strings.yml[0m:13: [91mrisky-file-permissions[0m [2mFile permissions unset or incorrect[0m 
[34mroles/pre_translation/tasks/extract_strings.yml[0m:17: [91myaml[0m [2mtrailing spaces[0m [2;91m(trailing-spaces)[0m 
[34mroles/pre_translation/tasks/extract_strings.yml[0m:27: [91mrisky-file-permissions[0m [2mFile permissions unset or incorrect[0m 
[34mroles/pre_translation/tasks/extract_strings.yml[0m:35: [91myaml[0m [2mwrong indentation: expected 4 but found 2[0m [2;91m(indentation)[0m 
[34mroles/pre_translation/tasks/extract_strings.yml[0m:40: [91mcommand-instead-of-shell[0m [2mUse shell only when shell functionality is required[0m 
[34mroles/pre_translation/tasks/extract_strings.yml[0m:45: [91mno-changed-when[0m [2mCommands should not change things if nothing needs doing[0m 
[34mroles/pre_translation/tasks/extract_strings.yml[0m:49: [91myaml[0m [2mtrailing spaces[0m [2;91m(trailing-spaces)[0m 
[34mroles/pre_translation/tasks/extract_strings.yml[0m:50: [91myaml[0m [2mtrailing spaces[0m [2;91m(trailing-spaces)[0m 
[34mroles/pre_translation/tasks/extract_strings.yml[0m:51: [91myaml[0m [2mno new line character at the end of file[0m [2;91m(new-line-at-end-of-file)[0m 
[34mroles/pre_translation/tasks/extract_strings.yml[0m:51: [91myaml[0m [2mtrailing spaces[0m [2;91m(trailing-spaces)[0m 
[34mroles/pre_translation/tasks/main.yml[0m:19: [91myaml[0m [2mtrailing spaces[0m [2;91m(trailing-spaces)[0m 
[34mroles/pre_translation/tasks/main.yml[0m:22: [91myaml[0m [2mtrailing spaces[0m [2;91m(trailing-spaces)[0m 
[34mroles/pre_translation/tasks/main.yml[0m:23: [91myaml[0m [2mno new line character at the end of file[0m [2;91m(new-line-at-end-of-file)[0m 
[34mroles/pre_translation/tasks/upload_strings_to_memsource.yml[0m:10: [91myaml[0m [2mtrailing spaces[0m [2;91m(trailing-spaces)[0m 
[34mroles/pre_translation/tasks/upload_strings_to_memsource.yml[0m:13: [91munnamed-task[0m [2mAll tasks should be named[0m 
[34mroles/pre_translation/tasks/upload_strings_to_memsource.yml[0m:15: [91myaml[0m [2mtrailing spaces[0m [2;91m(trailing-spaces)[0m 
[34mroles/pre_translation/tasks/upload_strings_to_memsource.yml[0m:16: [91munnamed-task[0m [2mAll tasks should be named[0m 
[34mroles/pre_translation/tasks/upload_strings_to_memsource.yml[0m:19: [91munnamed-task[0m [2mAll tasks should be named[0m 
[34mroles/pre_translation/tasks/upload_strings_to_memsource.yml[0m:25: [91myaml[0m [2mwrong indentation: expected 4 but found 2[0m [2;91m(indentation)[0m 
[34mroles/pre_translation/tasks/upload_strings_to_memsource.yml[0m:46: [91mno-log-password[0m [2mpassword should not be logged.[0m 
Loading module memsource_import_settings 
Linting module memsource_import_settings via flake8... 
Loading module memsource_import_settings_info 
Linting module memsource_import_settings_info via flake8... 
Loading module memsource_job 
Linting module memsource_job via flake8... 
Loading module memsource_job_info 
Linting module memsource_job_info via flake8... 
Loading module memsource_job_targetfile 
Linting module memsource_job_targetfile via flake8... 
plugins/modules/memsource_job_targetfile.py:69:1: F401 'ansible_collections.ansible.memsource.plugins.module_utils.memsource.get_action' imported but unused 
Loading module memsource_project 
Linting module memsource_project via flake8... 
Loading module memsource_project_info 
Linting module memsource_project_info via flake8... 
plugins/modules/memsource_project_info.py:35:12: W291 trailing whitespace 
Loading module memsource_project_template_info 
Linting module memsource_project_template_info via flake8... 
Loading module_utils memsource 
Linting module_utils memsource via flake8... 
Loading doc_fragments memsource 
Linting doc_fragments memsource via flake8... 
Collection loading complete 
Processing via galaxy-importer complete 
Checking dependencies in importer data 
Publishing collection 
Collection published 
Import completed with 66 warnings and 0 errors 
```

**Comment by adityamulik** [comment]
Hello @Spredzy.

Thanks for raising the issue, there had been a lot of indentation issues which was identified as per the trace log shared above. I have opened a PR, wherein I have added the GitHub action for Ansible-Lint and it's currently a WIP, so the PR is in draft mode. I will move it to be available for merge once all the lint issues are resolved.

Ref: https://github.com/ansible/ansible-collection-memsource/pull/36
